server/src/queries.rs

```diff
@@ -0,0 +1,252 @@
+use crate::router_utils::{
+    estimate_flight_time_minutes, get_nearby_nodes, get_route, init_router, Aircraft,
+    NearbyLocationQuery, RouteQuery, LANDING_AND_UNLOADING_TIME_MIN, LOADING_AND_TAKEOFF_TIME_MIN,
+    NODES, SAN_FRANCISCO,
+};
+use crate::scheduler_grpc::{
+    CancelFlightResponse, ConfirmFlightResponse, FlightPriority, FlightStatus, Id, QueryFlightPlan,
+    QueryFlightRequest, QueryFlightResponse,
+};
+use std::collections::HashMap;
+
+use crate::calendar_utils::{Calendar, Tz};
+use chrono::{Duration, NaiveDateTime, TimeZone};
+use once_cell::sync::OnceCell;
+use prost_types::Timestamp;
+use std::str::FromStr;
+use std::sync::Mutex;
+use std::time::SystemTime;
+use svc_storage_client_grpc::client::{
+    storage_rpc_client::StorageRpcClient, AircraftFilter, FlightPlan, VertiportFilter,
+};
+use tokio;
+use tonic::{Request, Response, Status};
+use uuid::Uuid;
+
+const CANCEL_FLIGHT_SECONDS: u64 = 30;
+
+fn unconfirmed_flight_plans() -> &'static Mutex<HashMap<String, FlightPlan>> {
+    static INSTANCE: OnceCell<Mutex<HashMap<String, FlightPlan>>> = OnceCell::new();
+    INSTANCE.get_or_init(|| Mutex::new(HashMap::new()))
+}
+
+fn cancel_flight_after_timeout(id: String) {
+    tokio::spawn(async move {
+        tokio::time::sleep(core::time::Duration::from_secs(CANCEL_FLIGHT_SECONDS)).await;
+        let mut flight_plans = unconfirmed_flight_plans().lock().unwrap();
+        if flight_plans.get(&id).is_some() {
+            flight_plans.remove(&id);
+        };
+    });
+}
+
+///Finds the first possible flight for customer location, flight type and requested time.
+pub async fn query_flight(
+    request: Request<QueryFlightRequest>,
+    mut storage_client: StorageRpcClient<tonic::transport::Channel>,
+) -> Result<Response<QueryFlightResponse>, Status> {
+    let flight_request = request.into_inner();
+    // 1. Fetch vertiports from customer request
+    let _r_vertiports = storage_client
+        .vertiports(Request::new(VertiportFilter {}))
+        .await?
+        .into_inner()
+        .vertiports;
+    //TODO use vertiports from DB instead of NODES
+    //2. Find route and cost between requested vertiports
+    if NODES.get().is_none() {
```

Comment:
Indeed all of this should be handled by lib-router, should be basically no mention of nodes in svc-scheduler

Reply:
When we have test data from svc-storage, we will just call init_router with vertiports so there will be no nodes. So this is pending svc-storage PR first